iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp

```diff
@@ -354,9 +355,48 @@ void DispatchRegionOp::getCanonicalizationPatterns(
   results.insert<ClosureOptimizationPattern<DispatchRegionOp>>(context);
 }
 
+/// Turns a flow.dispatch.workgroups op that just performs buffer fill into
+/// a flow.tensor.splat op.
+struct ConvertPureFillDispatchIntoTensorSplat final
+    : public OpRewritePattern<DispatchWorkgroupsOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(DispatchWorkgroupsOp dispatchOp,
+                                PatternRewriter &rewriter) const override {
+    // Make sure we just have one block.
+    Block &entryBlock = dispatchOp.getClosureBodyRegion().front();
+    if (!isa<Flow::ReturnOp>(entryBlock.getTerminator())) return failure();
+
+    // Make sure we just have one block argument.
+    if (entryBlock.getNumArguments() != 1) return failure();
+    BlockArgument entryArg = entryBlock.getArgument(0);
+
+    // Make sure we are only storing to the tensor of the block argument,
+    // at the full range.
+    if (!entryArg.hasOneUse()) return failure();
+    auto storeOp = dyn_cast<DispatchTensorStoreOp>(*entryArg.user_begin());
+    if (!storeOp || !storeOp.offsets().empty() || !storeOp.sizes().empty() ||
```

Comment:
style: we follow google convention here and use {} around multi-line ifs please (I know LLVM doesn't require them, but they are wrong ;)